examples/example-gateway/clients/baz/baz.go

```diff
@@ -34,23 +31,42 @@ func NewClient(config *zanzibar.StaticConfig, gateway *zanzibar.Gateway) *BazCli
 	port := config.MustGetInt("clients.baz.port")
 	sc.Peers().Add(ip + ":" + strconv.Itoa(int(port)))
 
-	client := zt.NewClient(gateway.Channel, serviceName)
+	client := zanzibar.NewClient(gateway.Channel, serviceName)
+
+	// TODO: (lu) maybe set these at per method level
+	timeout := time.Duration(config.MustGetInt("clients.baz.timeout")) * time.Millisecond
+	timeoutPerAttempt := time.Duration(config.MustGetInt("clients.baz.timeoutPerAttempt")) * time.Millisecond
 
 	return &BazClient{
-		thriftService: serviceName,
-		client:        client,
+		thriftService:     serviceName,
+		client:            client,
+		timeout:           timeout,
+		timeoutPerAttempt: timeoutPerAttempt,
 	}
 }
 
 // BazClient is the client to talk to SimpleService backend.
 type BazClient struct {
 	thriftService string
-	client        zt.TChanClient
+	client        zanzibar.TChanClient
+
+	timeout           time.Duration
+	timeoutPerAttempt time.Duration
 }
 
 // Call ...
 func (c *BazClient) Call(ctx context.Context, reqHeaders map[string]string, args *baz.SimpleService_Call_Args) (map[string]string, *baz.BazResponse, error) {
 	var result baz.SimpleService_Call_Result
+
+	retryOpts := &tchannel.RetryOptions{
```

Comment:
We should move this logic into the `Call()` function. 

That way generated code does not import tchannel, only `zanzibar/runtime`.